src/mysql/MySQL.chpl

```diff
@@ -0,0 +1,677 @@
+                throw new QueryExecutionError();
+            }
+
+            else if ((this._cptr_result == c_nil) && (mysql_errno(this._cptr_mysqlconn) == 0)) {
+                // Result is nil. Should it be? Check using mysql_field_count (as per example in API docs)
+                if (mysql_field_count(this._cptr_mysqlconn) == 0) {
+                    this._nFields = 0;
+                }
+                else {
+                    writeln(createStringWithNewBuffer(mysql_error(this._cptr_mysqlconn)));
+                    throw new QueryExecutionError();
+                }
+            }
+
+            // If all is fine and there is some result:
+            else if ((this._cptr_result != c_nil) && (mysql_errno(this._cptr_mysqlconn) == 0)) {
+                this._nFields = mysql_num_fields(this._cptr_result): int(32);
+                this._cptr_fields = mysql_fetch_fields(this._cptr_result);
+                
+                this._nRows = mysql_num_rows(this._cptr_result);
+            }
+
+            else {
+                writeln("[Error] Error occurred while storing result.");
+                throw new QueryExecutionError();
+                writeln(createStringWithNewBuffer(mysql_error(this._cptr_mysqlconn)));
+            }
+        }
+
+        /*
+        Execute a batch of SQL statements in a single transaction.
+        Commits after their execution, or rolls back if there was an error.
+            :arg statements: an array of :class:`Statement` objects to be executed
+            :type statements: array of :class:`Statement` objects
+        */
+        override proc executeBatch(statements: [] owned Statement) throws {
+            this._connection.beginTransaction();
+            var hasError: bool = false;
+
+            // TODO: memory management
+            for stmt in statements {
+                try {
+                    this.execute(stmt);
+                }
+                catch err: QueryExecutionError {
+                    hasError = true;
+                    this._connection.rollback();
+
+                    // re-throw the query execution error
+                    throw new QueryExecutionError();
+                }
+            }
+
+            if (!hasError) {
+                this._connection.commit();
+            }
+        }
+
+        /*
+        Fetches the next row of the result set.
+        Returns a row whcih gives false on calling isValid() on it.
+        */
+        override proc fetchone(): owned MySQLRow? {
+            if (this._curRow >= this._nRows) {
+                return nil;
+            }
+
+            var nextRow: MYSQL_ROW = mysql_fetch_row(this._cptr_result);
+            // TODO: is there any way to check if nextRow is NULL here?
+            // using nextRow == nil or nextRow == c_nil don't work
+
+            this._curRow += 1;
+
+            // init the row:
+            var _row = new MySQLRow(nextRow, this._cptr_fields, true);
```

Comment:
As mentioned in `MySQLRow` do we even need valid? As I can see you never used `false`

Reply:
You're right, we don't need that anymore. I'll update this.